lib/rawHttpRequest/validateAndProcessBody.js

```diff
@@ -0,0 +1,12 @@
+const validateBody = require('./validateBody.js');
+const processBody = require('./processBody.js');
+
+module.exports = async function (params) {
+	if (validateBody(params)) {
+		return processBody(params.body)
+		.then((body) => {
```

Comment:
Could you use await here since it's an async function i.e. `params.processedBody = await processBody(params.body)`? Otherwise I don't think it needs to be an async function.

Reply:
I primarily made this an async function so that when it is used via utils, the user has to always use the function as async/promise. Otherwise without the async, the if stmt creates a situation when in one case a promise is returned, and the other case something synchronous, i.e. `undefined` - and so there's an async vs sync situation that the user may not be aware of. So by making this async, it's always an async situation in terms of usage.

But anyhow - I'll change this to use await.